packages/roosterjs-content-model-api/lib/modelApi/common/queryContentModelBlocks.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+    ContentModelBlockType,
+    ReadonlyContentModelBlock,
+    ReadonlyContentModelBlockGroup,
+} from 'roosterjs-content-model-types';
+
+/**
+ * Query content model blocks
+ * @param group The block group to query
+ * @param blockType The type of block to query @default 'Paragraph'
+ * @param filter Optional selector to filter the blocks
+ * @param findFirstOnly True to return the first block only, false to return all blocks
+ */
+export function queryContentModelBlocks<T extends ReadonlyContentModelBlock>(
+    group: ReadonlyContentModelBlockGroup,
+    blockType?: ContentModelBlockType,
+    filter?: (element: T) => element is T,
+    findFirstOnly?: boolean
+): T[] {
+    const type = blockType || 'Paragraph';
+    const elements: T[] = [];
+    for (let i = 0; i < group.blocks.length; i++) {
+        if (findFirstOnly && elements.length > 0) {
+            return elements;
+        }
+        const block = group.blocks[i];
+        const results = queryContentModelBlocksInternal<T>(block, type, filter, findFirstOnly);
+        elements.push(...results);
+    }
+    return elements;
+}
+
+function queryContentModelBlocksInternal<T extends ReadonlyContentModelBlock>(
```

Comment:
since you don't need the parameter `options` now, you can merge these two functions together, so no need to duplicate the for loop for block group